include/pisa/query/algorithm/range_taat_query.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "query/queries.hpp"
+#include "topk_queue.hpp"
+
+namespace pisa {
+
+template <typename QueryAlg>
+struct range_taat_query {
+
+    range_taat_query(topk_queue& topk)
+        : m_topk(topk) {}
+
+    template <typename CursorRange, typename Acc>
+    uint64_t operator()(CursorRange &&cursors, uint64_t max_docid, size_t range_size, Acc &&accumulator)
+    {
+
+        if (cursors.empty()) {
+            return 0;
+        }
+
+        accumulator.init();
+
+        for (size_t end = range_size;
+             end + range_size <= max_docid; end += range_size) {
+            process_range(cursors, end, accumulator);
+        }
+        process_range(cursors, max_docid, accumulator);
```

Comment:
I think we'll end up processing twice the last range if the total length is a multiplier of the range size. I think the `for` condition should be instead `end + range_size < max_docid`, and that should fix it.